docker-compose.yml

```diff
@@ -3,6 +3,8 @@ version: '3'
 volumes:
   ## Kludge for not having the ./docker directory bound recursively
   dockerdirectory:
```

Comment:
we may not need this as we have `mysql_data` from this PR

Reply:
I have never understood why this is here … thanks for asking 😁 

I did some investigation to find out the current behaviour and what would happen if i removed.

### current behaviour – kludged `/plugins/woocommerce-payments/docker`
- The kludge prevents everything in `./docker` from being mapped into the container. 
- There's a lot of subfolders, logs etc, and these are mapped elsewhere. 
- So I think this kludge is to keep the plugin folder neat & tidy, and not have stuff redundantly mapped.

<img width="695" alt="Screenshot 2023-09-20 at 11 34 59 AM" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/e4596841-72a8-4111-8bf3-bd54a663078e">

### experiment: remove `dockerdirectory` volume, no kludge
- All the subfolders of `./docker` are mapped into `/plugins/woocommerce-payments/docker`.
- It's a bit messy and redundant.

So I think this kludge is worth keeping. I'll update the comment(s) to make the purpose clearer – TIL!

Reply:
The workaround remains, I've added comments and reordered mappings to make this clearer. 1222a9ee7a74a6632e357e76b6ebb81a54a31726